.github/workflows/linting.yml

```
@@ -9,29 +9,58 @@ on:
 jobs:
 
   flake8:
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        python-version: [3.6]
+    runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python ${{ matrix.python-version }}
+
+      - name: Set up Python
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: 3.9
 
       - name: Install dependencies
         run: |
           pip install flake8
-          pip install pyroma
 
       - name: Run linting
         run: |
           flake8 .
 
+  pyroma:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: |
+          pip install pyroma
+
       - name: Check package metadata
         run: |
           pyroma --min=10 .
+
+  mypy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: |
+          pip install mypy
+
+      - name: Check package metadata
+        run: |
+          mypy -m torch_geometric
```

mypy needs some type stub packages installed to work properly, the `--install-types --non-interactive` flags in combination are the appropriate way to do this in a CI setting. Not sure if `--ignore-missing-imports` is necessary or not. Typically, I usually set up mypy to work on the source files itself instead of using `-m`, so I'm not sure what'll happen next

```suggestion
          mypy --install-types --non-interactive --ignore-missing-imports -m torch_geometric
```

---

btw are you guys celebrating Karneval all the way up in Dortmund? Kölle alaaf ;)

---

Thanks for digging into this. I'm still a `mypy` noob :(
I applied the suggested changes, let's see how it goes.

PS: We are not really celebrating Karneval here in Dortmund :D

---

Looks like it worked, but now the real difficult part of mypy begins. The way I applied this to some of my big projects like PyKEEN was to only apply it to sub modules incrementally. It took a long time, and probably will motivate some pretty major refactoring

---

Super, thanks a lot. Indeed, we have now arrived at the challenging part. Going one submodule at a time sounds like a good idea. It also seems that `mypy` cannot handle decorators such as `@torch.jit._overload`, so I'm not even sure if I can resolve all of its complaints.